docs/templates/block-template-controller.md

```diff
@@ -0,0 +1,77 @@
+# BlockTemplateController
+[Source file](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/BlockTemplatesController.php)
+
+The BlockTemplateController class contains all the business logic which loads the templates into the Site Editor or on the front-end through various hooks available in WordPress & WooCommerce core. Without documenting every method individually, I will look to provide some insight into key functionality. 
+
+## Table of Contents
+
+* [Overview](#overview)
+* [add_block_templates( $query_result, $query, $template_type )](#add_block_templates-query_result-query-template_type-)
+* [get_block_file_template( $template, $id, $template_type )](#get_block_file_template-template-id-template_type-)
```

Comment:
Just double checking, is the ending `-` correct as an anchor?

Reply:
Yep, this is correct.